src/fft.rg

```diff
@@ -144,133 +152,435 @@ function fft.generate_fft_interface(itype, dtype)
+    format.println("Calling fftw_plan_dft to store fftw_plan in p.p...")
+
+    --R2C: Float to Complex32
+    if dtype_size == 8 and real_flag then
+      format.println("R2C: Float to Complex32: Not Supported")
+      --AK Note: I don't actually understand this that well, how the size of each dimension is stored in the 'n' array
+      var n : int[dim]
+      ;[data.range(dim):map(function(i) return rquote n[i] = hi.x[i] - lo.x[i] + 1 end end)] --n is a array of size dim storing the size of each dimension
+      format.println("calling fftwf_plan_dft_r2c")
+      --p.float_p = fftw_c.fftwf_plan_dft_r2c(dim, &n[0], [&float](input_base), [&fftw_c.fftwf_complex](output_base), fftw_c.FFTW_ESTIMATE)
+
+    elseif dtype_size == 8 then
+      format.println("R2C: Complex32 to Complex32: Not Supported")
+      var n : int[dim] 
+      ;[data.range(dim):map(function(i) return rquote n[i] = hi.x[i] - lo.x[i] + 1 end end)]
+      format.println("calling fftwf_plan_dft")
+      --p.float_p = fftw_c.fftwf_plan_dft(dim, &n[0], [&fftw_c.fftwf_complex](input_base), [&fftw_c.fftwf_complex](output_base), fftw_c.FFTW_FORWARD, fftw_c.FFTW_ESTIMATE)
+
+    elseif dtype_size == 16 and real_flag then
+      format.println("R2C: Double to Complex64: Calling fftw_plan_dft_r2c")
+      var n : int[dim]
+      ;[data.range(dim):map(function(i) return rquote n[i] = hi.x[i] - lo.x[i] + 1 end end)]
+      p.p = fftw_c.fftw_plan_dft_r2c(dim, &n[0], [&double](input_base), [&fftw_c.fftw_complex](output_base), fftw_c.FFTW_ESTIMATE)
+
+    elseif dtype_size == 16 then
+      format.println("R2C: Complex64 to Complex64: Calling fftw_plan_dft_r2c")
+      var n : int[dim]
+      ;[data.range(dim):map(function(i) return rquote n[i] = hi.x[i] - lo.x[i] + 1 end end)]
+      format.println("n[0] is {}, dim is {}", n[0], dim)
+      p.p = fftw_c.fftw_plan_dft(dim, &n[0], [&fftw_c.fftw_complex](input_base), [&fftw_c.fftw_complex](output_base), fftw_c.FFTW_FORWARD, fftw_c.FFTW_ESTIMATE)
+    end
+
+    p.address_space = address_space
+
+    --If GPUs are available, call make_plan_GPU
+    if gpu_available then
+      format.println("Num_local_gpus is {}", iface.get_num_local_gpus())
+      if iface.get_num_local_gpus() > 0 then
+        format.println("GPUs identified: calling make_plan_gpu...")
+        make_plan_gpu(input, output, plan, p.address_space)
+      end 
+    end
+  end
+
+
+   --Task: make_plan_batch in GPU version. Calls cufftPlanMany and stores plan in cufft_p
+  local make_plan_gpu_batch
+  if gpu_available then
+    __demand(__cuda, __leaf)
+
+    task make_plan_gpu_batch(input : region(ispace(itype), dtype_in), output : region(ispace(itype), dtype_out), plan : region(ispace(int1d), iface.plan), address_space : c.legion_address_space_t)
```

Comment:
So, I think this is actually not correct? We need an `itype` in N+1 dimensions.

Reply:
The way the API is set up is that the user will pass in a region that is size n, if the transform is of dimension n-1. That's the reason we need the 'n_batch' array. Does this address your comment?